lib_eio_linux/eio_stubs.c

```diff
@@ -33,10 +34,15 @@ CAMLprim value caml_eio_mkdirat(value v_fd, value v_path, value v_perm) {
 CAMLprim value caml_eio_getrandom(value v_ba, value v_off, value v_len) {
   CAMLparam1(v_ba);
   ssize_t ret;
-  void *buf = Caml_ba_data_val(v_ba) + Long_val(v_off);
-  caml_enter_blocking_section();
-  ret = getrandom(buf, Long_val(v_len), 0);
-  caml_leave_blocking_section();
+  ssize_t off = (ssize_t)Long_val(v_off);
+  ssize_t len = (ssize_t)Long_val(v_len);
+  do {
+    void *buf = Caml_ba_data_val(v_ba) + off;
+    caml_enter_blocking_section();
+    ret = getrandom(buf, len - off, 0);
```

Comment:
We just need the length here, don't we?
```suggestion
    ret = getrandom(buf, len, 0);
```